src/cs/Ssh/Events/SshReportProgressEventArgs.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.DevTunnels.Ssh.Events;
+
+/// <summary>
+/// Event raised to report connection progress.
+/// </summary>
+[DebuggerStepThrough]
+public class SshReportProgressEventArgs : EventArgs
+{
+	public SshReportProgressEventArgs(Progress progress, int? sessionNumber = null)
+	{
+		Progress = progress.ToString();
+		SessionNumber = sessionNumber;
+	}
+
+	/// <summary>
+	/// Specifies the progress event that is being reported. See <see cref="Progress"/>
+	/// for a description of the different progress events that can be reported.
+	/// </summary>
+	public string Progress { get; }
```

Comment:
Since this is using an enum, can the type be Progress

Reply:
Thanks, as discussed. I like that better.